network/framework/src/protocols/network/mod.rs

```diff
@@ -214,31 +262,52 @@ impl<TMessage> Stream for NetworkEvents<TMessage> {
     }
 }
 
+fn unix_micros() -> u64 {
```

Comment:
Any reason not to use time service? https://github.com/aptos-labs/aptos-core/blob/ce056658101865b91af44078941c8ea36458119d/crates/aptos-time-service/src/lib.rs#L155

Reply:
the usage sites are pretty awkward to plumb that through to